### ISSUE TITLE
Using "ctx.random.Shuffle" in typescript

### ISSUE THREAD

**Issue by jakehsiao** [boardgameio/boardgame.io#866]
Before adding typescript, just use "ctx.random.Shuffle" works well.
`G.deck = ctx.random.Shuffle(G.deck);`

But after adding typescript, "ctx.random" becomes "ctx.random?", and everytime I use the shuffle function, I must add some extra codes like:
`G.deck = ctx.random?.Shuffle(G.deck) || G.deck;`
That makes the code looks more redundant than before.

Are there any good methods to deal with this? I don't want every "ctx.random" method call becomes "ctx.random?" with "||" required at every call. I want typescript to know that ctx.random is guaranteed to be at here.

**Comment by larry801** [comment]
I wrapped shuffle in an utility function

```js
export const die = (ctx: Ctx, faces: number): number => {
    return ctx.random?.Die(faces) || faces;
}

export const shuffle = (ctx: Ctx, arg: any[]): any[] => {
    return ctx.random?.Shuffle(arg) || arg;
}
```

**Comment by delucis** [comment]
Ultimately, this will be fixed by #662 — the typing issue is because we use the `Ctx` type for both the data object available on the client and the object enhanced with random etc.

If you're concerned about the pointless checks in the compiled JavaScript you can also assert that random exists using `!`:

```ts
ctx.random!.Shuffle(arg)
```

Theoretically you lose some safety, but in moves random is guaranteed to be there. The one other issue is once #662 is implemented, `!` might trip you up while refactoring because at that point `random` won't be on `ctx` anymore.

**Comment by jakehsiao** [comment]
@delucis Thanks that's a good method.
